tests/persistence/test_persistence_leverage.py

```diff
@@ -0,0 +1,638 @@
+            = (275.97543219 * 0.00004099) + (275.97543219 * 0.00004099 * 0.0025)
+            = 0.01134051354788177
+    """
+    trade = Trade(
+        pair='ETH/BTC',
+        stake_amount=0.001,
+        amount=5,
+        open_rate=0.00004099,
+        open_date=datetime.utcnow() - timedelta(hours=0, minutes=10),
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        interest_rate=0.0005,
+        exchange='kraken',
+        leverage=3,
+        interest_mode=InterestMode.HOURSPER4
+    )
+    trade.open_order_id = 'open_trade'
+    trade.update(market_lev_buy_order)  # Buy @ 0.00001099
+    # Get the open rate price with the standard fee rate
+    assert trade._calc_open_trade_value() == 0.01134051354788177
+    trade.fee_open = 0.003
+    # Get the open rate price with a custom fee rate
+    assert trade._calc_open_trade_value() == 0.011346169664364504
+
+
+@pytest.mark.usefixtures("init_persistence")
+def test_calc_open_close_trade_price_lev(limit_lev_buy_order, limit_lev_sell_order, fee):
+    """
+        5 hour leveraged trade on Binance
+
+        fee: 0.25% base
+        interest_rate: 0.05% per day
+        open_rate: 0.00001099 base
+        close_rate: 0.00001173 base
+        amount: 272.97543219 crypto
+        stake_amount: 0.0009999999999226999 base
+        borrowed: 0.0019999999998453998  base
+        time-periods: 5 hours(rounds up to 5/24 time-period of 1 day)
+        interest: borrowed * interest_rate * time-periods
+                    = 0.0019999999998453998 * 0.0005 * 5/24 = 2.0833333331722917e-07 base
+        open_value: (amount * open_rate) + (amount * open_rate * fee)
+            = (272.97543219 * 0.00001099) + (272.97543219 * 0.00001099 * 0.0025)
+            = 0.0030074999997675204
+        close_value: ((amount_closed * close_rate) - (amount_closed * close_rate * fee)) - interest
+            = (272.97543219 * 0.00001173)
+                - (272.97543219 * 0.00001173 * 0.0025)
+                - 2.0833333331722917e-07
+            = 0.003193788481706411
+        stake_value: (amount/lev * open_rate) + (amount/lev * open_rate * fee)
+            = (272.97543219/3 * 0.00001099) + (272.97543219/3 * 0.00001099 * 0.0025)
+            = 0.0010024999999225066
+        total_profit =  close_value - open_value
+            = 0.003193788481706411 - 0.0030074999997675204
+            = 0.00018628848193889044
+        total_profit_percentage = total_profit / stake_value
+            = 0.00018628848193889054 / 0.0010024999999225066
+            = 0.18582392214792087
+    """
+    trade = Trade(
+        pair='ETH/BTC',
+        stake_amount=0.0009999999999226999,
+        open_rate=0.01,
+        amount=5,
+        open_date=datetime.utcnow() - timedelta(hours=4, minutes=55),
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        exchange='binance',
+        interest_rate=0.0005,
+        interest_mode=InterestMode.HOURSPERDAY
+    )
```

Comment:
How's that test work?
While it seems to assume and test leverage - leverage is not actually set... so based on this - i'm not quite certain what it actually tries to test.

Also adding different leverage values don't have any effect on the result (so a trade with leverage=3 and leverage=5 produces the same final profit).

Reply:
The interest value would be different. The leverage is set in `tests/conf.py`